src/tests/class_tests/openms/source/FileTypes_test.cpp

```diff
@@ -147,4 +149,13 @@ START_SECTION([EXTRA] FileTypes::FileTypeList)
 
   END_SECTION
 
+  START_SECTION(static FileTypes::FileTypeList typesWithProperties(const std::vector<FileProperties>& features))
+  {
+    std::vector<FileTypes::FileProperties> f;
+    f.push_back(FileTypes::FileProperties::READABLE);
+    FileTypeList g = FileTypeList::typesWithProperties(f);
+    TEST_EQUAL(g.getTypes().size(), 2);
```

Comment:
this needs extending.

1) test that empty filters return the full list
2) the full list should be equal to the size of the known FileTypes
3) The full list should contain each file type exactly once (otherwise its possible to pass the test for 2) by using duplicating one type and omitting another)